components/base/sensorcontrolled/movestraight.go

```diff
@@ -101,6 +118,18 @@ func (sb *sensorBase) MoveStraight(
 				if err != nil {
 					return err
 				}
+			} else {
+				currTime := time.Now()
+				vels, err := sb.velocities.LinearVelocity(ctx, nil)
+				if err != nil {
+					return err
+				}
+				deltaTime := currTime.Sub(prevTime).Seconds()
+				// calculate the estimated change in position based on the latest velocity
+				deltaPosMm := sign(mmPerSec) * vels.Y * deltaTime * 1000
```

Comment:
Why use `vels.Y` while ignoring the other components? If I mistakenly mount my movement sensor crooked, won't that mess everything up? I would have gotten the total magnitude of the velocity (including X, Y, and Z components).

Reply:
the control loop/base is currently designed assuming the base is a differential drive base, so only control in the Y direction and we assume the Y direction will be used for forward/backward movement. any movement in the X or Z directions is currently not controlled by the base. 

I also am assuming that the movement sensor reports velocity wrt the robot's frame, though a user could easily mount a sensor incorrectly, and the rest of the sensorcontrolled base would fail at that point. We could see this as another argument that more components/services need access to the frame system tbh

Reply:
Very good response! Let's leave as-is, and do stuff with the frame system later if it becomes important.